addon/adapters/batch-request.js

```diff
@@ -22,31 +27,31 @@ export default JSONAPIAdapter.extend({
   },
 
   batchCreate(items, options) {
-    if (isPresent(options) && isPresent(options.skipStoreUpdate)) {
-      return this._batch(items, 'POST', options.skipStoreUpdate);
-    }
+    options = options || {};
+    let httpMethod = options.httpMethod || 'POST';
+    let skipStoreUpdate = options.skipStoreUpdate || false;
 
-    return this._batch(items, 'POST', false);
+    return this._batch(items, ACTION_NAME.CRAETE, httpMethod, skipStoreUpdate, options.getIdsFunc);
   },
 
   batchUpdate(items, options) {
-    if (isPresent(options) && isPresent(options.skipStoreUpdate)) {
-      return this._batch(items, 'PATCH', options.skipStoreUpdate);
-    }
+    options = options || {};
+    let httpMethod = options.httpMethod || 'PATCH';
+    let skipStoreUpdate = options.skipStoreUpdate || false;
 
-    return this._batch(items, 'PATCH', false);
+    return this._batch(items, ACTION_NAME.UPDATE, httpMethod, skipStoreUpdate, options.getIdsFunc);
   },
 
   batchDelete(items) {
-    return this._batch(items, 'DELETE');
+    return this._batch(items, ACTION_NAME.DELETE, 'DELETE');
   },
 
-  _batch(items, actionName, skipStoreUpdate) {
+  _batch(items, actionName, httpMethod, skipStoreUpdate, getIdsFunc) {
```

Comment:
I don't like this method signature. `actionName` is actually an int from the `ACTION_NAME` hash. And passing both the code and the string (`httpMethod`) seems like duplication.